python_modules/dagster/dagster/_core/storage/partitions/base.py

```diff
@@ -0,0 +1,44 @@
+from abc import ABC, abstractmethod
+from typing import Callable, Optional, Sequence
+
+from dagster._core.instance import MayHaveInstanceWeakref
+
+
+class PartitionsStorage(ABC, MayHaveInstanceWeakref):
```

Comment:
Curious about @prha and/or @alangenfeld 's thoughts on adding `PartitionsStorage` vs. making this part of of `EventLogStorage`.

Reply:
Yeah... I was unsure about this one too. It didn't seem to make sense as part of event log storage as the partitions aren't generated from events, but I don't feel strongly about this.

Reply:
Let's just roll this into event log storage.  We're moving towards unifying all the storage abstractions anyway.